src/widgets/gl_widget.cpp

```diff
@@ -40,7 +41,28 @@ namespace QMapLibre {
 
 /*! Default constructor */
 GLWidget::GLWidget(const Settings &settings)
-    : d_ptr(std::make_unique<GLWidgetPrivate>(this, settings)) {}
+    : d_ptr(std::make_unique<GLWidgetPrivate>(this, settings)) {
+    // Ensure we get an OpenGL 3.2+ core profile to use MapLibre's modern
+    // renderer. This does nothing on platforms where such a context cannot be
+    // provided (e.g. WebAssembly), where Qt will fall back transparently.
+#ifdef __EMSCRIPTEN__
+    // WebAssembly: request a WebGL2 / OpenGL ES 3.0 context. Core profile is
```

Comment:
i started a bit looking into preserving the webgl support, if we move to OpenGL 3+. I don't know yet if there's throughput to webgl2, or if we might even use too modern OpenGL features for that.